### PR TITLE
More SF4 compatibility: CLI-commands

### LINKED ISSUE #161: No compatibility with SF4
#### Sonata packages

```
$ composer show --latest 'sonata-project/*'
sonata-project/admin-bundle              3.30.1             3.30.1             The missing Symfony Admin Generator
sonata-project/block-bundle              3.9.2              3.9.2              Symfony SonataBlockBundle
sonata-project/cache                     2.0.1              2.0.1              Cache library
sonata-project/classification-bundle     dev-master 485494c dev-master 485494c Symfony SonataClassificationBundle
sonata-project/core-bundle               3.9.0              3.9.0              Symfony SonataCoreBundle
sonata-project/datagrid-bundle           2.3.1              2.3.1              Symfony SonataDatagridBundle
sonata-project/doctrine-extensions       1.0.2              1.0.2              Doctrine2 behavioral extensions
sonata-project/doctrine-orm-admin-bundle 3.3.0              3.3.0              Symfony Sonata / Integrate Doctrine ORM into the SonataAdminBundle
sonata-project/easy-extends-bundle       2.3.0              2.3.0              Symfony SonataEasyExtendsBundle
sonata-project/exporter                  1.8.0              1.8.0              Lightweight Exporter library
sonata-project/translation-bundle        dev-master 785a2c9 dev-master 785a2c9 SonataTranslationBundle
```

#### Symfony packages

```
symfony/asset                v4.0.3             v4.0.3             Symfony Asset Component
symfony/browser-kit          v4.0.3             v4.0.3             Symfony BrowserKit Component
symfony/cache                v4.0.3             v4.0.3             Symfony Cache component with PSR-6, PSR-16, and tags
symfony/class-loader         v3.4.3             v3.4.3             Symfony ClassLoader Component
symfony/config               v4.0.3             v4.0.3             Symfony Config Component
symfony/console              v4.0.3             v4.0.3             Symfony Console Component
symfony/css-selector         v4.0.3             v4.0.3             Symfony CssSelector Component
symfony/debug                v4.0.3             v4.0.3             Symfony Debug Component
symfony/debug-bundle         v4.0.3             v4.0.3             Symfony DebugBundle
symfony/dependency-injection v4.0.3             v4.0.3             Symfony DependencyInjection Component
symfony/doctrine-bridge      v4.0.3             v4.0.3             Symfony Doctrine Bridge
symfony/dom-crawler          v4.0.3             v4.0.3             Symfony DomCrawler Component
symfony/dotenv               v4.0.3             v4.0.3             Registers environment variables from a .env file
symfony/event-dispatcher     v4.0.3             v4.0.3             Symfony EventDispatcher Component
symfony/expression-language  v4.0.3             v4.0.3             Symfony ExpressionLanguage Component
symfony/filesystem           v4.0.3             v4.0.3             Symfony Filesystem Component
symfony/finder               v4.0.3             v4.0.3             Symfony Finder Component
symfony/flex                 v1.0.61            v1.0.61           
symfony/form                 v4.0.3             v4.0.3             Symfony Form Component
symfony/framework-bundle     v4.0.3             v4.0.3             Symfony FrameworkBundle
symfony/http-foundation      v4.0.3             v4.0.3             Symfony HttpFoundation Component
symfony/http-kernel          v4.0.3             v4.0.3             Symfony HttpKernel Component
symfony/inflector            v4.0.3             v4.0.3             Symfony Inflector Component
symfony/intl                 v4.0.3             v4.0.3             A PHP replacement layer for the C intl extension that includes additional data from the ICU libr...
symfony/lts                  dev-master 396c5fc dev-master 396c5fc Enforces Long Term Supported versions of Symfony components
symfony/maker-bundle         v1.0.2             v1.0.2            
symfony/monolog-bridge       v4.0.3             v4.0.3             Symfony Monolog Bridge
symfony/monolog-bundle       v3.1.2             v3.1.2             Symfony MonologBundle
symfony/options-resolver     v4.0.3             v4.0.3             Symfony OptionsResolver Component
symfony/orm-pack             v1.0.5             v1.0.5             A pack for the Doctrine ORM
symfony/phpunit-bridge       v4.0.3             v4.0.3             Symfony PHPUnit Bridge
symfony/polyfill-intl-icu    v1.6.0             v1.6.0             Symfony polyfill for intl's ICU-related data and classes
symfony/polyfill-mbstring    v1.6.0             v1.6.0             Symfony polyfill for the Mbstring extension
symfony/polyfill-php72       v1.6.0             v1.6.0             Symfony polyfill backporting some PHP 7.2+ features to lower PHP versions
symfony/process              v4.0.3             v4.0.3             Symfony Process Component
symfony/profiler-pack        v1.0.3             v1.0.3             A pack for the Symfony web profiler
symfony/property-access      v4.0.3             v4.0.3             Symfony PropertyAccess Component
symfony/routing              v4.0.3             v4.0.3             Symfony Routing Component
symfony/security             v4.0.3             v4.0.3             Symfony Security Component
symfony/security-acl         v3.0.1             v3.0.1             Symfony Security Component - ACL (Access Control List)
symfony/security-bundle      v4.0.3             v4.0.3             Symfony SecurityBundle
symfony/stopwatch            v4.0.3             v4.0.3             Symfony Stopwatch Component
symfony/swiftmailer-bundle   v3.1.6             v3.1.6             Symfony SwiftmailerBundle
symfony/templating           v4.0.3             v4.0.3             Symfony Templating Component
symfony/translation          v4.0.3             v4.0.3             Symfony Translation Component
symfony/twig-bridge          v4.0.3             v4.0.3             Symfony Twig Bridge
symfony/twig-bundle          v4.0.3             v4.0.3             Symfony TwigBundle
symfony/validator            v4.0.3             v4.0.3             Symfony Validator Component
symfony/var-dumper           v4.0.3             v4.0.3             Symfony mechanism for exploring and dumping PHP variables
symfony/web-profiler-bundle  v4.0.3             v4.0.3             Symfony WebProfilerBundle
symfony/web-server-bundle    v4.0.3             v4.0.3             Symfony WebServerBundle
symfony/yaml                 v4.0.3             v4.0.3             Symfony Yaml Component
```

#### PHP version

```
PHP 7.2.1 (cli) (built: Jan  5 2018 17:34:14) ( NTS )
Copyright (c) 1997-2017 The PHP Group
Zend Engine v3.2.0, Copyright (c) 1998-2017 Zend Technologies
```

## Subject

I'm trying to get up and running with Sonata and it's components/extensions. One of the very nice features I wanted to use, was the SonataClassificationBundle. But following the installation instructions, I found this little problem: I wanted to run this command:
```
php bin/console sonata:easy-extends:generate --dest=src SonataClassificationBundle
```
but it keeps saying there are no commands defined in the sonata:easy-extends namespace.

## Steps to reproduce
```
php bin/console sonata:easy-extends:generate --dest=src SonataClassificationBundle
```

## Expected results
Success - or at least the command running

## Actual results
```
There are no commands defined in the "sonata:easy-extends" namespace.  
                                                                         
  Did you mean one of these?                                             
      sonata                                                             
      sonata:admin                                                       
      sonata:block                                                       
      sonata:core   

(with -vvv):

Exception trace:
 Symfony\Component\Console\Application->findNamespace() at /Library/WebServer/Documents/baert/vendor/symfony/console/Application.php:589
 Symfony\Component\Console\Application->find() at /Library/WebServer/Documents/baert/vendor/symfony/framework-bundle/Console/Application.php:93
 Symfony\Bundle\FrameworkBundle\Console\Application->find() at /Library/WebServer/Documents/baert/vendor/symfony/console/Application.php:216
 Symfony\Component\Console\Application->doRun() at /Library/WebServer/Documents/baert/vendor/symfony/framework-bundle/Console/Application.php:71
 Symfony\Bundle\FrameworkBundle\Console\Application->doRun() at /Library/WebServer/Documents/baert/vendor/symfony/console/Application.php:143
 Symfony\Component\Console\Application->run() at /Library/WebServer/Documents/baert/bin/console:39

```

### PR DESCRIPTION
**Pull request by BOAWEB**
I am targeting this branch, because this is the branch used for compatibility with SF4.

Closes #161 

## Changelog

```markdown
### Fixed
- Register commands explicitly in `commands.xml`
- Generator services are now public
```

## Subject

To be able to use this Bundle with SF4, especially the CLI-commands, there were some changes to be made. Commands defined were not showing up at all.

### CODE REVIEW

File: src/Resources/config/commands.xml [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+<container xmlns="http://symfony.com/schema/dic/services" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://symfony.com/schema/dic/services http://symfony.com/schema/dic/services/services-1.0.xsd">
+    <services>
+        <service id="Sonata\EasyExtendsBundle\Command\DumpMappingCommand" class="Sonata\EasyExtendsBundle\Command\DumpMappingCommand">
+            <tag name="console.command"/>
+        </service>
+        <service id="Sonata\EasyExtendsBundle\Command\GenerateCommand" class="Sonata\EasyExtendsBundle\Command\GenerateCommand">
+            <tag name="console.command"/>
+        </service>
+    </services>
+</container>
```
Comment (greg0ire) [inline src/Resources/config/commands.xml]: Please configure your IDE to always add a newline at end of file, this will make the red pictogram disappear.

**Comment by greg0ire** [comment]
Please associate your boa.be email address with your Github account, or change the email in your commits to an address already associated with it.

File: src/Resources/config/commands.xml (hunk 1 above)
Comment (BOAWEB) [reply src/Resources/config/commands.xml]: Done

**Comment by greg0ire** [comment]
For bonus points, squash your last commit with the commit that introduced the red pictogram

File: src/Resources/config/commands.xml [hunk 2]
```diff
@@ -0,0 +1,11 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<container xmlns="http://symfony.com/schema/dic/services" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://symfony.com/schema/dic/services http://symfony.com/schema/dic/services/services-1.0.xsd">
+    <services>
+        <service id="Sonata\EasyExtendsBundle\Command\DumpMappingCommand" class="Sonata\EasyExtendsBundle\Command\DumpMappingCommand">
```
Comment (jordisala1991) [inline src/Resources/config/commands.xml]: Console commands needs to be public or private? I am not sure. Do you know it @greg0ire ?

Comment (OskarStark) [reply src/Resources/config/commands.xml]: private IMO, as the are not injected

Comment (greg0ire) [reply src/Resources/config/commands.xml]: public seems less risky, See the sf slack

**Comment by greg0ire** [comment]
Thanks @BOAWEB

**Comment by jordisala1991** [comment]
If we said to make them public why did you merge before it was changed @greg0ire ? 😅

**Comment by greg0ire** [comment]
> If we said to make them public why did you merge before it was changed @greg0ire ? :sweat_smile:

WAT?

File: src/Resources/config/generator.xml [hunk 3]
```diff
@@ -1,10 +1,10 @@
...
-        <service id="sonata.easy_extends.generator.orm" class="Sonata\EasyExtendsBundle\Generator\OrmGenerator"/>
-        <service id="sonata.easy_extends.generator.odm" class="Sonata\EasyExtendsBundle\Generator\OdmGenerator"/>
-        <service id="sonata.easy_extends.generator.phpcr" class="Sonata\EasyExtendsBundle\Generator\PHPCRGenerator"/>
-        <service id="sonata.easy_extends.generator.serializer" class="Sonata\EasyExtendsBundle\Generator\SerializerGenerator"/>
+        <service id="sonata.easy_extends.generator.bundle" class="Sonata\EasyExtendsBundle\Generator\BundleGenerator" public="true"/>
+        <service id="sonata.easy_extends.generator.orm" class="Sonata\EasyExtendsBundle\Generator\OrmGenerator" public="true"/>
+        <service id="sonata.easy_extends.generator.odm" class="Sonata\EasyExtendsBundle\Generator\OdmGenerator" public="true"/>
+        <service id="sonata.easy_extends.generator.phpcr" class="Sonata\EasyExtendsBundle\Generator\PHPCRGenerator" public="true"/>
+        <service id="sonata.easy_extends.generator.serializer" class="Sonata\EasyExtendsBundle\Generator\SerializerGenerator" public="true"/>
```
Comment (greg0ire) [inline src/Resources/config/generator.xml]: @jordisala1991 oh my bad, I was confused by this, sorry...

**Pull request merged**